string file_name
bool is_url
